internal/core/key.go

```diff
@@ -626,6 +626,25 @@ func (k *IndexDataStoreKey) ToString() string {
 	return string(k.Bytes())
 }
 
+// IsEqual returns true if the two keys are equal
+func (k *IndexDataStoreKey) IsEqual(other IndexDataStoreKey) bool {
```

Comment:
nitpick: It looks a little odd to see an equality function taking a pointer and a value as it's two arguments, if feeling generous it might be nicer to change this so that `k` and `other` are of the same type :)

Reply:
I see what you mean, but it also feels weird to have all other receivers declared as pointers and only this one as value (if I change to all values). But if I change both to pointers, it won't be clear from the API, if we check deep equality of just pointers.
I don't have a strong opinion here either, but would tend more toward keeping it as is, as it has slightly better performance (because of the pointer), and has clear API.

Reply:
> but would tend more toward keeping it as is

Not a problem, is only a nitpick, and will vary from person to person which weirdness they tolerate more :)